src/EditorFeatures/Core.Wpf/InlineParamNameHints/InlineParamNameHintsTagger.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.ExtractMethod;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+{
+    /// <summary>
+    /// The purpose of this tagger is to convert the InlineParamNameHintsDataTags to
+    /// the InlineParamNameHintsTag, which actually creates the UIElement. It reacts to
+    /// tags changing and updates the adornments accordingly.
```

Comment:
crefs when mentioning the other types in our codebase, so that way refactorings don't break this.